packages/select/src/common/itemRenderer.ts

```diff
@@ -36,18 +36,28 @@ export type IItemRendererProps = ItemRendererProps;
 /**
  * An object describing how to render a particular item.
  * An `itemRenderer` receives the item as its first argument, and this object as its second argument.
+ *
+ * Make sure to forward the provided `ref` to the rendered element (usually via the `elementRef` prop on `MenuItem`/`MenuItem2`)
+ * to ensure that scrolling to active items works correctly.
+ *
+ * @template T type of the DOM element to attach a ref to, usually the component's root container (defaults to MenuItem's HTMLLIElement)
+ * @template U type of the DOM element to attach a click handler to, usually the component's root container (defaults to MenuItem's HTMLAnchorElement)
  */
-export interface ItemRendererProps {
-    /** A ref that receives the native HTML element rendered by this item. */
-    ref: Ref<any>;
+export interface ItemRendererProps<T extends HTMLElement = HTMLLIElement, U extends HTMLElement = HTMLAnchorElement> {
```

Comment:
we need two type params because MenuItems are weird and attach their ref to a different DOM element than the click handler. we should probably reconcile that at some point 🤷🏽

Reply:
actually, this turned out to be a bad approach. see https://github.com/palantir/blueprint/pull/5946